packages/webview_flutter/webview_flutter_wkwebview/pigeons/web_kit.dart

```diff
@@ -735,6 +756,18 @@ abstract class WKUIDelegateFlutterApi {
     WKFrameInfoData frame,
     WKMediaCaptureTypeData type,
   );
+
+  /// Callback to Dart function `WKUIDelegate.runJavaScriptDialog`.
+  @ObjCSelector(
+    'runJavaScriptDialogForDelegateWithIdentifier:type:message:defaultText:',
+  )
+  @async
+  JavaScriptDialogCompletionData runJavaScriptDialog(
```

Comment:
Since the native `WKUIDelegate` separates handling dialogs into separate methods, this should also.